src/core/network/injectSendNetworkRequest.js

```diff
@@ -12,38 +12,37 @@ governing permissions and limitations under the License.
 
 import { stackError } from "../../utils";
 
-export default ({ logger, networkStrategy, isRetryableHttpStatusCode }) => {
+export default ({
+  logger,
+  sendFetchRequest,
+  sendBeaconRequest,
+  isRetryableHttpStatusCode
+}) => {
   /**
    * Send a network request and returns details about the response.
-   *
-   * @param {Object} payload This will be JSON stringified and sent as the post body.
-   * @param {String} url The URL to which the request should be sent.
-   * @param {String} requestID A unique ID for the request.
    */
-  return ({ payload, url, requestId }) => {
-    const stringifiedPayload = JSON.stringify(payload);
-
+  return ({ requestId, url, payload, useSendBeacon }) => {
     // We want to log raw payload and event data rather than
     // our fancy wrapper objects. Calling payload.toJSON() is
     // insufficient to get all the nested raw data, because it's
     // not recursive (it doesn't call toJSON() on the event objects).
     // Parsing the result of JSON.stringify(), however, gives the
     // fully recursive raw data.
-    // JSON.parse is expensive so we short circuit if logging is disabled.
-    if (logger.enabled) {
-      logger.logOnBeforeNetworkRequest({
-        url,
-        requestId,
-        payload: JSON.parse(stringifiedPayload)
```

Comment:
The idea behind wrapping this particular statement in `if (logger.enabled) {` was so that we would only run `JSON.parse` if we were actually going to log the result to the console (because it was considered expensive). It's not really helpful though, because we end up running `JSON.parse(stringifiedPayload)` eventually within this module anyway. So now we just do it once up front.